File: tests/fixtures/padded.rs
// Struct with obvious padding waste — used by integration tests.
#[repr(C)]
pub struct Padded {
    pub a: u8,
    pub b: u64,
    pub c: u8,
}

// Well-ordered struct — no padding, no findings expected.
#[repr(C)]
pub struct Clean {
    pub b: u64,
    pub a: u8,
    pub c: u8,
}

File: tests/fixtures/padded.zig
// Struct with obvious padding waste — used by integration tests.
const Padded = struct {
    a: u8,
    b: f64,
    c: u8,
};

// Struct that is already optimally ordered.
const Tight = struct {
    b: f64,
    a: u8,
    c: u8,
};
